From 09996b3e9169ec71dc18a942c1ab5cded171dd7e Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Mon, 28 May 2012 23:28:22 -0700
Subject: [PATCH 279/859] ASoC: fsi: bugfix: ensure dma is terminated

FSI DMAEngine has to be stopped certainly at the start/stop time.
Without this patch, it will include noise on playback.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Signed-off-by: Mark Brown <broonie@opensource.wolfsonmicro.com>
(cherry picked from commit fbe42f66c66c279b4ed9b8f515058a09bc731c49)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 sound/soc/sh/fsi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/sh/fsi.c b/sound/soc/sh/fsi.c
index 837c042..3fc81b3 100644
--- a/sound/soc/sh/fsi.c
+++ b/sound/soc/sh/fsi.c
@@ -1179,6 +1179,8 @@ static void fsi_dma_push_start_stop(struct fsi_priv *fsi, struct fsi_stream *io,
 
 	fsi_reg_mask_set(fsi, OUT_DMAC, DMA_ON, enable);
 
+	dmaengine_terminate_all(io->chan);
+
 	if (fsi_is_clk_master(fsi))
 		fsi_master_mask_set(master, CLK_RST, clk, (enable) ? clk : 0);
 }
-- 
1.8.1.4

